### PR TITLE
Feature/spot hash

### PR DESCRIPTION
**Pull request by anguyen698**
# Administrative Info
Make sure your branch name conforms to: `feature/hotfix/.../describe_your_branch_with_underscores

### Changes
What changes did you make?
- Added insertSpotHash()
- Added update() to lotModel
- Added route for update()

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tested my program thoroughly
- [ ] Did you add/sub fields in the environment file? if so please update .env_sample

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

insertSpotHash code
insert-spot

update code
update-func

update-route

### CODE REVIEW

**Review by Eldar827 — APPROVED** [review]
lgtm

**Review by samuelng678 — APPROVED** [review]
lgtm

File: database/model/spotsModel.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const db = require('../dbConfig');
+
+/**
+ * insert spot hash using the information given to us
+ * @param spot_hash
+ */
+ async function insertSpotHash (spot_hash){
+    try {
+        await db('spots')
+            .insert(spot_hash);
```
Comment (YuyusanK) [inline database/model/spotsModel.js:10]: change insert to update based off of id.

**Review by YuyusanK — CHANGES_REQUESTED** [review]

File: database/model/lotModel.js [hunk 2]
```diff
@@ -18,10 +18,23 @@ async function insert (parking_lot_info){
...
+
+/**
+ * update parking lot using the information given to us
+ * @param id
+ * @param changes
+ */
+ async function update (id, changes){
+    const [result] = await db('lots')
+        .where({ id })
+        .update({ changes });
```
Comment (YuyusanK) [inline database/model/lotModel.js:34]: `changes` is going to be passed through as a json. Please take out `{ }`.

**Review by YuyusanK — CHANGES_REQUESTED** [review]

File: database/model/lotModel.js [hunk 3]
```diff
@@ -18,10 +18,23 @@ async function insert (parking_lot_info){
...
+    }
+}
+
+/**
+ * update parking lot using the information given to us
+ * @param id
+ * @param changes
+ */
+ async function update (id, changes){
+    const [result] = await db('lots')
```
Comment (YuyusanK) [inline database/model/lotModel.js:32]: please take this out. `const [result] =`

Comment (YuyusanK) [inline database/model/lotModel.js:32]: the whole update function should be in a try-catch

**Review by YuyusanK — CHANGES_REQUESTED** [review]

File: database/model/spotsModel.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const db = require('../dbConfig');
+
+/**
+ * insert spot hash using the information given to us
```
Comment (YuyusanK) [inline database/model/spotsModel.js:4]: description is not correct

File: database/model/spotsModel.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ * @param spot_id
+ * @param spot_hash
+ */
+ async function update (spot_id, spot_hash){
+    try {
+        await db('spots')
+            .where(spot_id)
```
Comment (YuyusanK) [inline database/model/spotsModel.js:11]: use `{ }` to create a json for `spot_id`

**Review by YuyusanK — CHANGES_REQUESTED** [review]

File: routes/parking_lot.js [hunk 6]
```diff
@@ -23,6 +23,21 @@ router.post("/parking_lot", async function(req, res) {
...
 
+/* PUT parking_lot route */
+router.put("/parking_lot", async function (req, res) {
+    const {id} = req.params;
+    const changes = req.body;
 
+    try {
+        const result = await lot_model.update(id, changes);
+        if (result) {
+            res.status(200).json({updated: result})
```
Comment (YuyusanK) [inline routes/parking_lot.js:34]: `updated` should be `message`

**Review by YuyusanK — CHANGES_REQUESTED** [review]

**Review by YuyusanK — APPROVED** [review]

**Pull request merged**